--- libewf-20080501/libewf/Makefile.am.orig	2008-04-12 10:03:30.000000000 +0200
+++ libewf-20080501/libewf/Makefile.am	2008-07-18 18:04:40.130034707 +0200
@@ -50,7 +50,7 @@
 	libewf_file_io_handle.c libewf_file_io_handle.h \
 	libewf_file_io_pool.c libewf_file_io_pool.h
 
-libewf_la_LIBADD  = ../common/libcommon.la
+libewf_la_LIBADD  = ../common/libcommon.la $(LIBZ)
 libewf_la_LDFLAGS = -no-undefined -version-info 1:3:0
 
 distclean: clean
--- libewf-20080501/configure.ac.orig	2008-05-01 14:31:31.000000000 +0200
+++ libewf-20080501/configure.ac	2008-07-18 18:04:26.918035545 +0200
@@ -247,7 +247,7 @@
 AC_CHECK_LIB(
  z,
  adler32,
- [],
+ [LIBZ=-lz; AH_CHECK_LIB([z]) AC_DEFINE([HAVE_LIBZ])],
  [AC_MSG_FAILURE(
   [Missing function: adler32 in library: zlib.],
   [1])])
@@ -276,7 +276,7 @@
   [Missing function: uncompress in library: zlib.],
   [1])])
 
-AC_CHECK_LIB( crypto, MD5_Init, [], [ ac_libewf_have_libcrypto_md5=no ] )
+AC_CHECK_LIB( crypto, MD5_Init, [LIBCRYPTO=-lcrypto; AH_CHECK_LIB([crypto]) AC_DEFINE([HAVE_LIBCRYPTO])], [ ac_libewf_have_libcrypto_md5=no ] )
 AC_CHECK_LIB( crypto, MD5_Update, [ ac_libewf_dummy=yes ], [ ac_libewf_have_libcrypto_md5=no ] )
 AC_CHECK_LIB( crypto, MD5_Final, [ ac_libewf_dummy=yes ], [ ac_libewf_have_libcrypto_md5=no ] )
 
@@ -295,7 +295,7 @@
 AC_CHECK_LIB(
  uuid,
  uuid_generate,
- [],
+ [LIBUUID=-luuid; AH_CHECK_LIB([uuid]) AC_DEFINE([HAVE_LIBUUID])],
  [ ac_libewf_have_libuuid=no ])
 
 AC_CHECK_LIB(
@@ -314,6 +314,10 @@
   [1],
   [Defines if uuid_generate_time funtion is present in libuuid.])])
 
+AC_SUBST([LIBCRYPTO])
+AC_SUBST([LIBUUID])
+AC_SUBST([LIBZ])
+
 dnl Check for headers
 AC_CHECK_HEADER(
  [sys/types.h],
--- libewf-20080501/ewftools/Makefile.am.orig	2008-03-18 21:19:09.000000000 +0100
+++ libewf-20080501/ewftools/Makefile.am	2008-07-18 19:11:20.614039735 +0200
@@ -17,7 +17,7 @@
 	ewfsha1.h \
 	ewfacquire.c
 
-ewfacquire_LDADD  = ../common/libcommon.la ../libewf/libewf.la
+ewfacquire_LDADD  = ../common/libcommon.la ../libewf/libewf.la $(LIBUUID) $(LIBCRYPTO)
 
 ewfacquirestream_SOURCES = \
 	ewfbyte_size_string.c ewfbyte_size_string.h \
@@ -34,7 +34,7 @@
 	ewfsha1.h \
 	ewfacquirestream.c
 
-ewfacquirestream_LDADD  = ../common/libcommon.la ../libewf/libewf.la
+ewfacquirestream_LDADD  = ../common/libcommon.la ../libewf/libewf.la $(LIBUUID) $(LIBCRYPTO)
 
 ewfalter_SOURCES = \
 	ewfbyte_size_string.c ewfbyte_size_string.h \
@@ -51,7 +51,7 @@
 	ewfsha1.h \
 	ewfalter.c
 
-ewfalter_LDADD  = ../common/libcommon.la ../libewf/libewf.la
+ewfalter_LDADD  = ../common/libcommon.la ../libewf/libewf.la $(LIBUUID) $(LIBCRYPTO)
 
 ewfexport_SOURCES = \
 	ewfbyte_size_string.c ewfbyte_size_string.h \
@@ -68,7 +68,7 @@
 	ewfsha1.h \
 	ewfexport.c
 
-ewfexport_LDADD  = ../common/libcommon.la ../libewf/libewf.la
+ewfexport_LDADD  = ../common/libcommon.la ../libewf/libewf.la $(LIBUUID) $(LIBCRYPTO)
 
 ewfinfo_SOURCES = \
 	ewfbyte_size_string.c ewfbyte_size_string.h \
@@ -84,7 +84,7 @@
 	ewfsha1.h \
 	ewfinfo.c
 
-ewfinfo_LDADD  = ../common/libcommon.la ../libewf/libewf.la
+ewfinfo_LDADD  = ../common/libcommon.la ../libewf/libewf.la $(LIBUUID) $(LIBCRYPTO)
 
 ewfverify_SOURCES = \
 	ewfbyte_size_string.c ewfbyte_size_string.h \
@@ -100,7 +100,7 @@
 	ewfsha1.h \
 	ewfverify.c
 
-ewfverify_LDADD  = ../common/libcommon.la ../libewf/libewf.la
+ewfverify_LDADD  = ../common/libcommon.la ../libewf/libewf.la $(LIBUUID) $(LIBCRYPTO)
 
 distclean: clean
 	/bin/rm -f Makefile
