From cc617ecd58c766245888089079913c68bbc11109 Mon Sep 17 00:00:00 2001
From: Joachim Metz <joachim.metz@gmail.com>
Date: Mon, 16 Jul 2018 15:05:05 +0200
Subject: [PATCH] Changes to file name sanitation in export handle

---
 ewftools/export_handle.c | 104 +++++++++++++++++++++++++--------------
 1 file changed, 68 insertions(+), 36 deletions(-)

diff --git a/ewftools/export_handle.c b/ewftools/export_handle.c
index 22f3d7c0..89aea2ec 100644
--- a/ewftools/export_handle.c
+++ b/ewftools/export_handle.c
@@ -5890,11 +5890,13 @@ int export_handle_export_single_files(
      log_handle_t *log_handle,
      libcerror_error_t **error )
 {
-	libewf_file_entry_t *file_entry = NULL;
-	static char *function           = "export_handle_export_single_files";
-	size_t export_path_size         = 0;
-	int result                      = 0;
-	int status                      = PROCESS_STATUS_COMPLETED;
+	libewf_file_entry_t *file_entry    = NULL;
+	system_character_t *sanitized_name = NULL;
+	static char *function              = "export_handle_export_single_files";
+	size_t export_path_length          = 0;
+	size_t sanitized_name_size         = 0;
+	int result                         = 0;
+	int status                         = PROCESS_STATUS_COMPLETED;
 
 	if( export_handle == NULL )
 	{
@@ -5918,18 +5920,22 @@ int export_handle_export_single_files(
 
 		return( -1 );
 	}
-	export_path_size = 1 + system_string_length(
-	                        export_handle->target_path );
+	export_path_length = system_string_length(
+	                      export_handle->target_path );
 
 #if defined( HAVE_WIDE_SYSTEM_CHARACTER )
-	if( libcpath_path_sanitize_wide(
+	if( libcpath_path_get_sanitized_path_wide(
 	     export_handle->target_path,
-	     &export_path_size,
+	     export_path_length,
+	     &sanitized_name,
+	     &sanitized_name_size,
 	     error ) != 1 )
 #else
-	if( libcpath_path_sanitize(
+	if( libcpath_path_get_sanitized_path(
 	     export_handle->target_path,
-	     &export_path_size,
+	     export_path_length,
+	     &sanitized_name,
+	     &sanitized_name_size,
 	     error ) != 1 )
 #endif
 	{
@@ -5991,11 +5997,11 @@ int export_handle_export_single_files(
 	}
 #if defined( HAVE_WIDE_SYSTEM_CHARACTER )
 	if( libcpath_path_make_directory_wide(
-	     export_handle->target_path,
+	     sanitized_name,
 	     error ) != 1 )
 #else
 	if( libcpath_path_make_directory(
-	     export_handle->target_path,
+	     sanitized_name,
 	     error ) != 1 )
 #endif
 	{
@@ -6005,21 +6011,21 @@ int export_handle_export_single_files(
 		 LIBCERROR_IO_ERROR_WRITE_FAILED,
 		 "%s: unable to create directory: %" PRIs_SYSTEM "",
 		 function,
-		 export_handle->target_path );
+		 sanitized_name );
 
 		goto on_error;
 	}
 	log_handle_printf(
 	 log_handle,
 	 "Created directory: %" PRIs_SYSTEM ".\n",
-	 export_handle->target_path );
+	 sanitized_name );
 
 	result = export_handle_export_file_entry(
 	          export_handle,
 	          file_entry,
-	          export_handle->target_path,
-	          export_path_size,
-	          export_path_size - 1,
+	          sanitized_name,
+	          sanitized_name_size,
+	          sanitized_name_size - 1,
 	          log_handle,
 	          error );
 
@@ -6034,6 +6040,11 @@ int export_handle_export_single_files(
 
 		goto on_error;
 	}
+	memory_free(
+	 sanitized_name );
+
+	sanitized_name = NULL;
+
 	if( export_handle->abort != 0 )
 	{
 		status = PROCESS_STATUS_ABORTED;
@@ -6099,6 +6110,11 @@ int export_handle_export_single_files(
 		 &file_entry,
 		 NULL );
 	}
+	if( sanitized_name != NULL )
+	{
+		memory_free(
+		 sanitized_name );
+	}
 	return( -1 );
 }
 
@@ -6114,14 +6130,16 @@ int export_handle_export_file_entry(
      log_handle_t *log_handle,
      libcerror_error_t **error )
 {
-	system_character_t *name        = NULL;
-	system_character_t *target_path = NULL;
-	static char *function           = "export_handle_export_file_entry";
-	size_t name_size                = 0;
-	size_t target_path_size         = 0;
-	uint8_t file_entry_type         = 0;
-	int result                      = 0;
-	int return_value                = 0;
+	system_character_t *name           = NULL;
+	system_character_t *sanitized_name = NULL;
+	system_character_t *target_path    = NULL;
+	static char *function              = "export_handle_export_file_entry";
+	size_t name_size                   = 0;
+	size_t sanitized_name_size         = 0;
+	size_t target_path_size            = 0;
+	uint8_t file_entry_type            = 0;
+	int result                         = 0;
+	int return_value                   = 0;
 
 	if( export_handle == NULL )
 	{
@@ -6222,14 +6240,18 @@ int export_handle_export_file_entry(
 			goto on_error;
 		}
 #if defined( HAVE_WIDE_SYSTEM_CHARACTER )
-		if( libcpath_path_sanitize_filename_wide(
+		if( libcpath_path_get_sanitized_filename_wide(
 		     name,
-		     &name_size,
+		     name_size - 1,
+		     &sanitized_name,
+		     &sanitized_name_size,
 		     error ) != 1 )
 #else
-		if( libcpath_path_sanitize_filename(
+		if( libcpath_path_get_sanitized_filename(
 		     name,
-		     &name_size,
+		     name_size - 1,
+		     &sanitized_name,
+		     &sanitized_name_size,
 		     error ) != 1 )
 #endif
 		{
@@ -6242,14 +6264,19 @@ int export_handle_export_file_entry(
 
 			goto on_error;
 		}
+		memory_free(
+		 name );
+
+		name = NULL;
+
 #if defined( HAVE_WIDE_SYSTEM_CHARACTER )
 		if( libcpath_path_join_wide(
 		     &target_path,
 		     &target_path_size,
 		     export_path,
 		     export_path_size - 1,
-		     name,
-		     name_size - 1,
+		     sanitized_name,
+		     sanitized_name_size - 1,
 		     error ) != 1 )
 #else
 		if( libcpath_path_join(
@@ -6257,8 +6284,8 @@ int export_handle_export_file_entry(
 		     &target_path_size,
 		     export_path,
 		     export_path_size - 1,
-		     name,
-		     name_size - 1,
+		     sanitized_name,
+		     sanitized_name_size - 1,
 		     error ) != 1 )
 #endif
 		{
@@ -6272,9 +6299,9 @@ int export_handle_export_file_entry(
 			goto on_error;
 		}
 		memory_free(
-		 name );
+		 sanitized_name );
 
-		name = NULL;
+		sanitized_name = NULL;
 
 		if( target_path == NULL )
 		{
@@ -6443,6 +6470,11 @@ int export_handle_export_file_entry(
 		memory_free(
 		 target_path );
 	}
+	if( sanitized_name != NULL )
+	{
+		memory_free(
+		 sanitized_name );
+	}
 	if( name != NULL )
 	{
 		memory_free(
